src/osdp_cp.c

```diff
@@ -1387,60 +1387,53 @@ static int cp_detect_connection_topology(struct osdp *ctx)
 	return 0;
 }
 
-static struct osdp *__cp_setup(int num_pd, const osdp_pd_info_t *info_list)
-{
-	int i;
-	struct osdp_pd *pd = NULL;
-	struct osdp *ctx;
-	const osdp_pd_info_t *info;
-	char name[24] = {0};
-
-	ctx = calloc(1, sizeof(struct osdp));
-	if (ctx == NULL) {
-		LOG_PRINT("Failed to allocate osdp context");
-		return NULL;
-	}
-
-	input_check_init(ctx);
-
-	ctx->pd = calloc(1, sizeof(struct osdp_pd) * num_pd);
-	if (ctx->pd == NULL) {
-		LOG_PRINT("Failed to allocate osdp_pd[] context");
-		goto error;
-	}
-	ctx->_num_pd = num_pd;
-
-	for (i = 0; i < num_pd; i++) {
-		info = info_list + i;
-		pd = osdp_to_pd(ctx, i);
-		pd->idx = i;
-		pd->osdp_ctx = ctx;
-		if (info->name) {
-			strncpy(pd->name, info->name, OSDP_PD_NAME_MAXLEN - 1);
-		} else {
-			snprintf(pd->name, OSDP_PD_NAME_MAXLEN, "PD-%d", info->address);
-		}
-		pd->baud_rate = info->baud_rate;
-		pd->address = info->address;
-		pd->flags = info->flags;
-		pd->seq_number = -1;
-		SET_FLAG(pd, PD_FLAG_SC_DISABLED);
-		memcpy(&pd->channel, &info->channel, sizeof(struct osdp_channel));
-		if (info->scbk != NULL) {
-			memcpy(pd->sc.scbk, info->scbk, 16);
-			SET_FLAG(pd, PD_FLAG_HAS_SCBK);
-			CLEAR_FLAG(pd, PD_FLAG_SC_DISABLED);
-		} else if (is_enforce_secure(pd)) {
-			LOG_PRINT("SCBK must be passed for each PD when"
-				  " ENFORCE_SECURE is requested.");
-			goto error;
-		}
-		if (cp_cmd_queue_init(pd)) {
-			goto error;
-		}
-		if (IS_ENABLED(CONFIG_OSDP_SKIP_MARK_BYTE)) {
-			SET_FLAG(pd, PD_FLAG_PKT_SKIP_MARK);
-		}
+static int cp_add_pd(struct osdp *ctx, int new_num_pd, const osdp_pd_info_t *new_info_list) {
+    int i, old_num_pd;
+    struct osdp_pd *new_pd_array, *pd;
+    const osdp_pd_info_t *info;
+	char name[24] = { 0 };
+
+    old_num_pd = ctx->_num_pd;
+    new_pd_array = realloc(ctx->pd, sizeof(struct osdp_pd) * (old_num_pd + new_num_pd));
```

Comment:
realloc() does not provide zero-ed memory and this rest of the method makes a lot of default assumptions pegged on that belief.

This could be the reason why CI is failing.

Reply:
The clang-format that we have is dated; needs a bit of work before we can restart using it.

Reply:
For now, please set "indent using tabs" and "tabs are 8 spaces" in whatever editor you are using (or better, install editorconfig plugin and these will be automatically set for you). 

After that, a simple reformat of this section should fix most of the issues.

Reply:
Okay, I fixed your branch locally as clang-format is broken.